src/hooks/starsDependencies.js

```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import { allProducts } from "../components/common/allProducts";
+import { filter_products } from "../redux/actions/disProdsActions";
+
+const stars = [
+  { id: 5, status: false },
+  { id: 4, status: false },
+  { id: 3, status: false },
+  { id: 2, status: false },
+  { id: 1, status: false },
+];
+
+export function useStarsDependencies(props) {
+  const products = allProducts;
+  const dispatch = useDispatch();
+  const [ratingStars, setRatingStars] = useState(stars);
+  const starPattern = [
+    [1, 1, 1, 1, 1],
+    [1, 1, 1, 1, 0],
+    [1, 1, 1, 0, 0],
+    [1, 1, 0, 0, 0],
+    [1, 0, 0, 0, 0],
+  ];
+
+  function handleChange(e) {
```

Comment:
you can use useCallback here to declare the function, when you write the function like this every time the component rerender it will create new function, with useCallaback the function will created each time the dependency array changed